File: pipeline/kite9-pipeline-common/src/main/java/com/kite9/k9server/adl/holder/meta/MetaRead.java
package com.kite9.k9server.adl.holder.meta;

import java.net.URI;
import java.util.Map;

public interface MetaRead {
	
	String getTitle();

	URI getTopicUri();

	URI getUri();

	/**
	 * Information about the editing, creation etc. of the document.
	 */
	Map<String, Object> getMetaData();

}

File: pipeline/kite9-pipeline-common/src/main/kotlin/com/kite9/k9server/adl/holder/meta/MetaRead.kt
package com.kite9.k9server.adl.holder.meta

import com.kite9.k9server.uri.URI


interface MetaRead {

    val title: String
    val topicUri: URI
    val uri: URI

    /**
     * Information about the editing, creation etc. of the document.
     */
    val metaData: Map<String, Any>
}